### PR TITLE
Add 'str from' subcommand

### PR DESCRIPTION
**Pull request by bailey-layzer**
Correction of #2093, addresses #2033.

### CODE REVIEW

**Comment by fdncred** [comment]
@bailey-layzer Glad to see your PR is back. We're still debating internally how we'd like this to work. 
* One option is to leave it as is. 
* Another option is to make it a subcommand on `str` like `str from`. An example might be `ls | get size | math sum | str from` where `str from` is one of these, `str from --precise 2` and `str from --with-commas` and `str from --precise 2 --with-commas`.
* Yet another option, which may be beyond the scope of this PR, is to have a report command like `ls | report --settings here`. I'm not sure exactly what `--settings` looks like. This is also more broad in scope because `report` is taking all columns and doing something with them. Not just the numbers.

@bailey-layzer Do you have any strong preference?

@thegedge, @jonathandturner , @andrasio Any other thoughts on this PR?

**Comment by thegedge** [comment]
`str` is getting pretty loaded, but I think it's the best candidate. Some reasons for `str` over separate commands:

- Maintains coherence because it's basically the place where we convert arbitrary values to strings, sometimes with additional processing.
- Aggregates string processing functionality into a single "namespace", which helps with discoverability.

Precision makes sense as a flag. I could see `humanize` being a subcommand, but okay with it being a flag too for a general "from" subcommand.

**Comment by bailey-layzer** [comment]
`str` seems most natural to me, but given I'm just getting into this project I would be happy to go along with whichever option you all prefer.

**Comment by fdncred** [comment]
@bailey-layzer Good point about `--with-commas`. We've chatted a bit and if you're still up for it, we'd like you to transform this PR into `str from` with `--group-digits` in place of `--with-commas` and `--decimals` in place of `--precision`. There's still some debate. Naming is hard.

**Comment by fdncred** [comment]
Looks like you're making progress. Keep up the good work!

File: crates/nu-cli/Cargo.toml [hunk 1]
```diff
@@ -98,6 +99,12 @@ trash = {version = "1.0.1", optional = true}
 [target.'cfg(unix)'.dependencies]
 users = "0.10.0"
 
+# TODO this will be possible with new dependency resolver
+#   (currently on nightly behind -Zfeatures=itarget):
+#   https://github.com/rust-lang/cargo/issues/7914
+#[target.'cfg(not(windows))'.dependencies]
```
Comment (bailey-layzer) [inline crates/nu-cli/Cargo.toml:105]: Unfortunately, num-format needs Clang to format per system locale on Windows, which seems disproportionate for this functionality alone. num-format puts system locale behind a feature flag in Cargo.toml - currently, Cargo's dependency resolver can't differentiate features by system. Fortunately, there's a new dependency resolver (#https://github.com/rust-lang/cargo/issues/7914) which will enable this should it reach stable.

Comment (fdncred) [reply crates/nu-cli/Cargo.toml:105]: Ya, I noticed your TODO comments. What does that mean for this functionality now? We can't compile without clang? Also, I noticed the CI failed but that looks like a bogus error. Not sure what's going on there.

Comment (bailey-layzer) [reply crates/nu-cli/Cargo.toml:105]: Right now it just uses `en`, so it works but won't actually localize - with those lines uncommented it wasn't compiling on the Windows azure jobs.

**Comment by fdncred** [comment]
Can you push some meaningless change to see if that straightens out the CI?

File: crates/nu-cli/Cargo.toml (hunk 1 above)
Comment (fdncred) [reply crates/nu-cli/Cargo.toml:105]: ok, thanks.

**Comment by bailey-layzer** [comment]
Since I wrote a function to trim non-whitespace chars I can add a flag for that to `str trim` - wasn't sure if I should include that here or make a new PR.

**Comment by fdncred** [comment]
Nice - CI passed. @thegedge Would you mind reviewing this please?

**Comment by jonathandturner** [comment]
Taking a look now

File: crates/nu-cli/Cargo.toml (hunk 1 above)
Comment (jonathandturner) [reply crates/nu-cli/Cargo.toml:105]: This seems good for now. Once it lands, we should file an issue for follow-up so we can track the work on cargo.

**Comment by jonathandturner** [comment]
I like this approach. It feels like it will also keep us flexible for future expansion as we add support for other types, locales, etc.

Good work!

**Pull request merged**